src/webots/control/WbController.cpp

```diff
@@ -783,6 +783,12 @@ void WbController::startPython() {
 }
 
 void WbController::startMatlab() {
+#ifdef __linux__
+  if (qgetenv("SNAP_NAME") == "webots") {
```

Comment:
That's a detail, but we should probably create a `bool isSnap()` function in WbSysInfo (similar to the already existing `isRootUser` function) as we use this test at several places in the code.